runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -212,31 +220,57 @@ public long startCheckpoint() {
                 ImmutableMap.copyOf(this.mdkv);
         List<Long> continuationAddresses = new ArrayList<>();
 
+        if (enablePutAll) {
+            Iterable<List<Object>> partitions = Iterables.partition(map.keySet(), batchSize);
 
-        Iterable<List<Object>> partitions = Iterables.partition(map.keySet(), batchSize);
+            for (List<Object> partition : partitions) {
+                Map tmp = new HashMap();
+                for (Object k : partition) {
+                    tmp.put(keyMutator.apply(k), valueMutator.apply(map.get(k)));
+                }
 
-        for (List<Object> partition : partitions) {
-            Map tmp = new HashMap();
-            for (Object k : partition) {
-                tmp.put(keyMutator.apply(k), valueMutator.apply(map.get(k)));
-            }
+                SMREntry smrEntry = new SMREntry("putAll", new Object[]{tmp}, serializer);
+                MultiSMREntry smrEntries = new MultiSMREntry();
+                smrEntries.addTo(smrEntry);
 
-            SMREntry smrEntry = new SMREntry("putAll", new Object[]{tmp}, serializer);
-            MultiSMREntry smrEntries = new MultiSMREntry();
-            smrEntries.addTo(smrEntry);
+                CheckpointEntry cp = new CheckpointEntry(CheckpointEntry.CheckpointEntryType.CONTINUATION,
+                        author, checkpointId, streamId, mdkv, smrEntries);
 
-            CheckpointEntry cp = new CheckpointEntry(CheckpointEntry.CheckpointEntryType.CONTINUATION,
-                    author, checkpointId, streamId, mdkv, smrEntries);
+                long pos = sv.append(Collections.singleton(checkpointStreamID), cp, null);
 
-            long pos = sv.append(Collections.singleton(checkpointStreamID), cp, null);
+                postAppendFunc.accept(cp, pos);
+                continuationAddresses.add(pos);
 
-            postAppendFunc.accept(cp, pos);
-            continuationAddresses.add(pos);
-
-            numEntries++;
-            // CheckpointEntry::serialize() has a side-effect we use
-            // for an accurate count of serialized bytes of SRMEntries.
-            numBytes += cp.getSmrEntriesBytes();
+                numEntries++;
+                // CheckpointEntry::serialize() has a side-effect we use
+                // for an accurate count of serialized bytes of SRMEntries.
+                numBytes += cp.getSmrEntriesBytes();
+            }
+        } else {
+            Iterators.partition(map.keySet().stream()
+                    .map(k -> {
+                        return new SMREntry("put",
+                                new Object[]{keyMutator.apply(k), valueMutator.apply(map.get(k))},
+                                serializer);
+                    }).iterator(), batchSize)
+                    .forEachRemaining(entries -> {
+                        MultiSMREntry smrEntries = new MultiSMREntry();
+                        for (int i = 0; i < ((List) entries).size(); i++) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "List". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1905)